analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/CustomAssertions.cs

```diff
@@ -0,0 +1,28 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using FluentAssertions.Primitives;
+
+namespace SonarAnalyzer.UnitTest;
+
+internal static class CustomAssertions
+{
+    public static AndWhichConstraint<ObjectAssertions, T> ShouldNotBeNull<T>(this T t) => t.Should().BeAssignableTo<T>();
```

Comment:
I don't like this idea because it introduced another `Should()` for something niche (and weird). Just inline it.

Reply:
The inlined version is worse due to the generic parameter. But ok, will do.